pkg/consensus/consensus.go

```diff
@@ -209,6 +209,33 @@ func (s *service) eventLoop() {
 	}
 }
 
+func (s *service) newPrepareRequest() payload.PrepareRequest {
+	sr, err := s.Chain.GetStateRoot(s.Chain.BlockHeight())
+	if err != nil {
+		return new(prepareRequest)
```

Comment:
Looks dangerous, not sure it's correct.

Reply:
It will lead to a change view anyway, wasn't sure how to handle this.